systemvm/patches/debian/config/opt/cloud/bin/cs/CsHelper.py

```diff
@@ -179,12 +179,12 @@ def get_hostname():
         return line.strip()
 
 
-def execute(command):
+def execute(command, wait=True):
     """ Execute command """
     logging.debug("Executing: %s" % command)
     p = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
-    result = p.communicate()[0]
-    return result.splitlines()
+    if wait:
+        return p.communicate()[0].splitlines()
```

Comment:
@jburwell < I think your request, while valid, goes beyond the scope of this patch which was made to fix a problem we have in production.

To wait for an amount of time, it would be required to spawn a separate thread, as the `timeout` argument was added in Python 3 (3.2 I think).